engine/engine.go

```diff
@@ -96,6 +96,14 @@ func (o Opts) getLogicalOptimizers() []logicalplan.Optimizer {
 	return optimizers
 }
 
+// QueryOpts implements promql.QueryOpts but allows to override more engine default options.
+type QueryOpts struct {
+	promql.QueryOpts
+
+	// DecodingConcurrency can be used to override the DecodingConcurrency engine setting.
+	DecodingConcurrency int
```

Comment:
Would this become a new query parameter in Thanos to make it overridable?

Reply:
Yeah we could do that if we wanted! It just seemed as good as a start as any. I have planned some followups for this and wanted to keep the PR somewhat scoped

Reply:
I.e. We could use this to communicate partial response strategy to remote engine too.

Reply:
Sounds good!